src/client.ts

```diff
@@ -4627,22 +4627,24 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             threadId = null;
         }
         const reason = opts?.reason;
+        let withRelTypes = opts?.with_rel_types ?? opts?.with_relations;
 
-        if (
-            opts?.with_relations &&
-            this.canSupport.get(Feature.RelationBasedRedactions) === ServerSupport.Unsupported
-        ) {
+        if (typeof withRelTypes === "string") {
+            withRelTypes = [withRelTypes];
+        }
+
+        if (withRelTypes && this.canSupport.get(Feature.RelationBasedRedactions) === ServerSupport.Unsupported) {
             throw new Error(
                 "Server does not support relation based redactions " +
                     `roomId ${roomId} eventId ${eventId} txnId: ${txnId} threadId ${threadId}`,
             );
         }
 
-        const withRelations = opts?.with_relations
+        const withRelTypesContent = withRelTypes
             ? {
                   [this.canSupport.get(Feature.RelationBasedRedactions) === ServerSupport.Stable
                       ? MSC3912_RELATION_BASED_REDACTIONS_PROP.stable!
```

Comment:
my understanding is that this line is currently unreachable, given that `Feature.RelationBasedRedactions` has no defined `matrixVersion` - is that correct?

Not necessarily asking you to change anything here. Though I do think it would be clearer to factor out a `propName` variable instead of putting the ternary condition inline.

Reply:
> my understanding is that this line is currently unreachable, given that Feature.RelationBasedRedactions has no defined matrixVersion - is that correct?

Yes

> Not necessarily asking you to change anything here. Though I do think it would be clearer to factor out a propName variable instead of putting the ternary condition inline.

Done